trytond/trytond/ir/ui/view.py

```diff
@@ -435,15 +446,60 @@ def write(cls, records, values, *args):
         ModelView._fields_view_get_cache.clear()
 
     @classmethod
-    def set_width(cls, model, fields):
+    def get_width(cls, model, width):
+        for screen_size in WIDTH_BREAKPOINTS:
+            if width >= screen_size:
+                break
+        else:
+            screen_size = 0
+
+        user = Transaction().user
+        records = cls.search([
+            ('user', '=', user),
+            ('model', '=', model),
+            ('screen_size', '=', screen_size),
+            ])
+
+        if not records:
+            records = cls.search([
+                ('user', '=', user),
+                ('model', '=', model),
+                ['OR',
+                    ('screen_size', '<=', screen_size),
+                    ('screen_size', '=', None),
+                    ],
+                ],
+                order=[
+                    ('screen_size', 'DESC NULLS LAST'),
+                    ('client', 'ASC NULLS LAST'),
+                    ])
+        widths = {}
+        for width in records:
+            if width.field not in widths:
+                widths[width.field] = width.width
+        return widths
+
+    @classmethod
+    def set_width(cls, model, fields, client, width):
```

Comment:
This could probably be optimized to avoid a `delete` / `create`, when a `write` will usually be ok.

Right now it takes ~20ms (on my machine) every time a column is resized